### PR TITLE
Updated VIA Support for DP60

### PR DESCRIPTION
**Pull request by paulgali**
Updated for VIA support since VIAs overhaul. 

## Description

- Changed ______ to KC_TRNS in \via\keymap.c
- Included Underglow settings on the rules.mk VIA directory
- Updated Product Name in config.h (formerly "Dumplings") 
EDIT:
- Added LAYOUT_all for via
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by paulgali** [comment]
image

RGB confirmed as configurable in VIA

**Comment by paulgali** [comment]
Final changes have been made. The board has been [tested in VIA](https://imgur.com/a/Um3YKfx) and is working well. KLE is [here](http://www.keyboard-layout-editor.com/##@@_x:15.75%3B&=0,12%0A%0A%0A1,1&=0,13%0A%0A%0A1,1%3B&@_y:0.25&x:2.75&c=%23777777%3B&=0,0&_c=%23cccccc%3B&=0,1&=0,2&=0,3&=0,4&=0,5&=0,6&=4,7&=0,7&=0,8&=0,9&=0,10&=0,11&_c=%23aaaaaa&w:2%3B&=0,13%0A%0A%0A1,0%3B&@_x:2.75&w:1.5%3B&=1,0&_c=%23cccccc%3B&=1,1&=1,2&=1,3&=1,4&=1,5&=1,6&=1,7&=1,8&=1,9&=1,10&=1,11&=1,12&_w:1.5%3B&=1,13%0A%0A%0A0,0&_x:2&w:1.25&h:2&w2:1.5&h2:1&x2:-0.25%3B&=2,13%0A%0A%0A0,1%3B&@_x:2.75&c=%23aaaaaa&w:1.75%3B&=2,0&_c=%23cccccc%3B&=2,1&=2,2&=2,3&=2,4&=2,5&=2,6&=2,7&=2,8&=2,9&=2,10&=2,11&_c=%23777777&w:2.25%3B&=2,13%0A%0A%0A0,0&_x:1&c=%23cccccc%3B&=2,12%0A%0A%0A0,1%3B&@_c=%23aaaaaa&w:1.25%3B&=3,0%0A%0A%0A2,1&_c=%23cccccc%3B&=3,1%0A%0A%0A2,1&_x:0.5&c=%23aaaaaa&w:2.25%3B&=3,0%0A%0A%0A2,0&_c=%23cccccc%3B&=3,2&=3,3&=3,4&=3,5&=3,6&=3,7&=3,8&=3,9&=3,10&=3,11&_c=%23aaaaaa&w:2.75%3B&=3,12%0A%0A%0A3,0&_x:0.5&w:1.75%3B&=3,12%0A%0A%0A3,1&_c=%23cccccc%3B&=3,13%0A%0A%0A3,1%3B&@_x:2.75&c=%23aaaaaa&w:1.25%3B&=4,0%0A%0A%0A4,0&_w:1.25%3B&=4,1%0A%0A%0A4,0&_w:1.25%3B&=4,2%0A%0A%0A4,0&_w:6.25%3B&=4,6%0A%0A%0A4,0&_w:1.25%3B&=4,10%0A%0A%0A4,0&_w:1.25%3B&=4,11%0A%0A%0A4,0&_w:1.25%3B&=4,12%0A%0A%0A4,0&_w:1.25%3B&=4,13%0A%0A%0A4,0%3B&@_y:0.25&x:2.75&w:1.5%3B&=4,0%0A%0A%0A4,1&=4,1%0A%0A%0A4,1&_w:1.5%3B&=4,2%0A%0A%0A4,1&_w:7%3B&=4,6%0A%0A%0A4,1&_w:1.5%3B&=4,11%0A%0A%0A4,1&=4,12%0A%0A%0A4,1&_w:1.5%3B&=4,13%0A%0A%0A4,1%3B&@_y:0.25&x:2.75&w:1.5&d:true%3B&=%0A%0A%0A4,2&=4,1%0A%0A%0A4,2&_w:1.5%3B&=4,2%0A%0A%0A4,2&_w:7%3B&=4,6%0A%0A%0A4,2&_w:1.5%3B&=4,11%0A%0A%0A4,2&=4,12%0A%0A%0A4,2&_w:1.5&d:true%3B&=%0A%0A%0A4,2)

**Comment by paulgali** [comment]
Changed default VIA layouts. Only a Quality of life Keymap update before this is merged. Final Change.

https://imgur.com/a/A2QwEbQ

File: keyboards/dp60/config.h [hunk 1]
```diff
@@ -11,7 +11,7 @@
 #define PRODUCT_ID      0x00BE
 #define DEVICE_VER      0x0001
 #define MANUFACTURER    astro
-#define PRODUCT         Dumplings
+#define PRODUCT         DP60 PCB // Dumplings
```
Comment (fauxpark) [inline keyboards/dp60/config.h:14]:
```suggestion
#define PRODUCT         DP60
```

File: keyboards/dp60/dp60.h [hunk 2]
```diff
@@ -9,6 +9,52 @@
 // This a shortcut to help you visually see your layout.
 // The first section contains all of the arguements
 // The second converts the arguments into a two-dimensional array
+//
+//This is an enhancement to enable VIA mapping for the DP60 PCB. Thanks to @sigprof in qmk discord for guidance
+//
```
Comment (fauxpark) [inline keyboards/dp60/dp60.h:14]:
```suggestion
```

File: keyboards/dp60/keymaps/via/keymap.c [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
-      _______,   _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
-      _______,        _______,_______,_______,_______,_______,_______,_______, _______,_______,_______,_______,_______,
-      _______,        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
-      _______,_______,_______,                        _______,                        _______,_______,_______,_______),
+    [1] = LAYOUT_all(
+      KC_GRV, KC_F1,  KC_F2,  KC_F3,  KC_F4,  KC_F5,  KC_F6,  KC_F7,  KC_F8,  KC_F9, KC_F10, KC_F11, KC_F12,KC_TRNS,KC_DEL,
+      RESET,   KC_TRNS,KC_UP,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_PAUS,KC_SLCK,KC_PSCR,KC_PGUP,KC_TRNS,KC_INS,
+      KC_TRNS,        KC_LEFT,KC_DOWN,KC_RIGHT,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS, KC_TRNS,KC_TRNS,KC_HOME,KC_END,KC_TRNS, KC_TRNS,
+      KC_TRNS,KC_TRNS,        KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_PGDN,KC_TRNS,KC_TRNS,
+      KC_TRNS,KC_TRNS,KC_TRNS,                        KC_TRNS,                        KC_TRNS,MO(2),TG(0),KC_TRNS),
```
Comment (fauxpark) [inline keyboards/dp60/keymaps/via/keymap.c:16]:
```suggestion
      KC_GRV, KC_F1,  KC_F2,  KC_F3,  KC_F4,  KC_F5,  KC_F6,  KC_F7,  KC_F8,  KC_F9, KC_F10, KC_F11, KC_F12,_______,KC_DEL,
      RESET,   _______,KC_UP,_______,_______,_______,_______,_______,KC_PAUS,KC_SLCK,KC_PSCR,KC_PGUP,_______,KC_INS,
      _______,        KC_LEFT,KC_DOWN,KC_RIGHT,_______,_______,_______,_______, _______,_______,KC_HOME,KC_END,_______, _______,
      _______,_______,        _______,_______,_______,_______,_______,_______,_______,_______,_______,KC_PGDN,_______,_______,
      _______,_______,_______,                        _______,                        _______,MO(2),TG(0),_______),
```

File: keyboards/dp60/keymaps/via/keymap.c [hunk 4]
```diff
@@ -1,31 +1,31 @@
...
+      KC_MUTE,        KC_MPRV,KC_VOLD,KC_MNXT,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS, KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS, KC_TRNS,
+      KC_TRNS, KC_TRNS,       RGB_TOG,RGB_MOD,RGB_RMOD,RGB_VAI,RGB_VAD,RGB_HUI,RGB_HUD,RGB_SAI,RGB_SAD,KC_TRNS,KC_TRNS,KC_TRNS,
+      KC_TRNS,KC_TRNS,KC_TRNS,                        KC_MPLY,                        KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS),
+
+    [3] = LAYOUT_all(
+      KC_TRNS, KC_TRNS,KC_TRNS,  KC_TRNS,KC_TRNS,  KC_TRNS,KC_TRNS,  KC_TRNS,KC_TRNS,  KC_TRNS,KC_TRNS, KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,
+      KC_TRNS,   KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,
+      KC_TRNS,        KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS, KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,
+      KC_TRNS, KC_TRNS,       KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,
+      KC_TRNS,KC_TRNS,KC_TRNS,                        KC_TRNS,                        KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS)
```
Comment (fauxpark) [inline keyboards/dp60/keymaps/via/keymap.c:30]:
```suggestion
      _______, _______,_______,  _______,_______,  _______,_______,  _______,_______,  _______,_______, _______,_______,_______,_______,
      _______,   _______,KC_VOLU,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
      KC_MUTE,        KC_MPRV,KC_VOLD,KC_MNXT,_______,_______,_______,_______, _______,_______,_______,_______,_______, _______,
      _______, _______,       RGB_TOG,RGB_MOD,RGB_RMOD,RGB_VAI,RGB_VAD,RGB_HUI,RGB_HUD,RGB_SAI,RGB_SAD,_______,_______,_______,
      _______,_______,_______,                        KC_MPLY,                        _______,_______,_______,_______),

    [3] = LAYOUT_all(
      _______, _______,_______,  _______,_______,  _______,_______,  _______,_______,  _______,_______, _______,_______,_______,_______,
      _______,   _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
      _______,        _______,_______,_______,_______,_______,_______,_______, _______,_______,_______,_______,_______,_______,
      _______, _______,       _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
      _______,_______,_______,                        _______,                        _______,_______,_______,_______)
```

File: keyboards/dp60/keymaps/via/rules.mk [hunk 5]
```diff
@@ -1,2 +1,5 @@
 VIA_ENABLE = yes
 LTO_ENABLE = yes
+RGBLIGHT_ENABLE = yes #enables underglow, but will not disable per key leds
+# BOOTMAGIC_ENABLE = lite #already enabled in base directory
```
Comment (fauxpark) [inline keyboards/dp60/keymaps/via/rules.mk:4]:
```suggestion
```

File: keyboards/dp60/rules.mk [hunk 6]
```diff
@@ -22,14 +22,14 @@ COMMAND_ENABLE = no         # Commands for debug and configuration
...
-NKRO_ENABLE = no # USB Nkey Rollover
+NKRO_ENABLE = yes # USB Nkey Rollover
 BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
 MIDI_ENABLE = no            # MIDI support
 UNICODE_ENABLE = no         # Unicode
 BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
 AUDIO_ENABLE = no           # Audio output on port C6
 FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
-#RGBLIGHT_ENABLE = yes       # Use RGB underglow light
+# RGBLIGHT_ENABLE = Yes       # Use RGB underglow light
```
Comment (fauxpark) [inline keyboards/dp60/rules.mk:32]:
```suggestion
RGBLIGHT_ENABLE = no        # Use RGB underglow light
```

File: keyboards/dp60/keymaps/via/keymap.c (hunk 4 above)
Comment (paulgali) [reply keyboards/dp60/keymaps/via/keymap.c:30]: @fauxpark Thanks for suggestions. Isn't KC_TRNS reccomended instead of _______ inside via's keymap.c?

Comment (fauxpark) [reply keyboards/dp60/keymaps/via/keymap.c:30]: It's exactly the same, `_______` just makes the keymap more readable.

Comment (paulgali) [reply keyboards/dp60/keymaps/via/keymap.c:30]: OK, awesome. Will commit now.

**Comment by paulgali** [comment]
@fauxpark all changes committed

File: keyboards/dp60/keymaps/via/rules.mk [hunk 7]
```diff
@@ -1,2 +1,4 @@
 VIA_ENABLE = yes
 LTO_ENABLE = yes
+RGBLIGHT_ENABLE = yes #enables underglow, but will not disable per key leds
+MOUSEKEY_ENABLE = no
```
[the remark this answers is not in the document]
Comment (paulgali) [reply keyboards/dp60/keymaps/via/rules.mk:4]: Will Commit this change. There isn't much of a difference, and when compiling its roughly 87% full.

**Pull request merged**